src/compiler/AST/Canonical.elm

```diff
@@ -49,9 +55,14 @@ var qualifier name =
         }
 
 
-lambda : VarName -> Expr -> Expr
+lambda : VarName -> LocatedExpr -> LocatedExpr
 lambda argument body =
-    Lambda
-        { argument = argument
-        , body = body
-        }
+    -- we use body's location
```

Comment:
Here we arbitrarily choose to keep the location of the body when optimizing. It should live in Optimization or be fixed.

Reply:
Body's location is not enough though, right?
```
\x -> x + 1
```
Ideally we'd like it to start with the `\` and end with the `1` but if we use body's location, the location will only contain `x + 1` if I understand this correctly.

Reply:
For this, I transfered it to Desugar, its only reference.